django/santropolFeast/member/views.py

```diff
@@ -886,15 +887,117 @@ def geolocateAddress(request):
     return JsonResponse({'latitude': lat, 'longtitude': long})
 
 
-def change_status(request, id):
-    if request.method == "POST":
-        client = get_object_or_404(Client, pk=id)
-        status = request.POST.get('status')
-        client.status = status
-        client.save()
+def clientStatusScheduler(request, pk):
+    client = get_object_or_404(Client, pk=pk)
+    return render(request, 'client/modal/change_status.html', {
+        'client': client,
+        'client_status': Client.CLIENT_STATUS,
+        'status_to': request.GET.get('status', Client.PAUSED),
+    })
+
+
+def clientStatusAlterOrSchedule(request, pk):
+    if request.method == 'POST':
+        client = get_object_or_404(Client, pk=pk)
+        status_to = request.POST.get('status_to')
+        reason = request.POST.get('reason', '')
+        start_date = request.POST.get('start_date', '')
+        end_date = request.POST.get('end_date', '')
+
+        # Three available possibilities
+        # 1 - Immediat status update
+        if start_date == '' and end_date == '':
+            client_get_oldstatus_display = client.get_status_display()
+            client.status = status_to
+            client.save()
+            msg = 'Instantly modify client status from {} to {}.'.format(
+                client_get_oldstatus_display,
+                client.get_status_display()
+            )
+            if (reason != ''):
+                msg += ' Reason: {}'.format(reason)
+            note = Note(
+                note=msg,
+                author=request.user,
+                client=client
+            )
+            note.save()
```

Comment:
This code could be moved to the Note model into a function, like _add_note(client)_